metadetect/lsst/tests/test_lsst_dm_configs.py

```diff
@@ -0,0 +1,33 @@
+from lsst.meas.algorithms import SourceDetectionConfig
+
+
+def test_detect_config():
+    expected_keys = [
+        'minPixels',
+        'isotropicGrow',
+        'combinedGrow',
+        'nSigmaToGrow',
+        'returnOriginalFootprints',
+        'thresholdValue',
+        'includeThresholdMultiplier',
+        'thresholdType',
+        'thresholdPolarity',
+        'adjustBackground',
+        'reEstimateBackground',
+        'background',
+        'tempLocalBackground',
+        'doTempLocalBackground',
+        'tempWideBackground',
+        'doTempWideBackground',
+        'nPeaksMaxSimple',
+        'nSigmaForKernel',
+        'statsMask',
+        'excludeMaskPlanes'
+    ]
+
+    config = SourceDetectionConfig()
+
+    keys = config.toDict().keys()
+
+    for key in keys:
+        assert key in expected_keys, f'found unexpected config key {key}'
```

Comment:
I am not sure what this is meant to achieve. It will fail the moment we add a new field to this config, but won't if we remove a config. DM guarantees to deprecate a field before removing and not remove a field until the next official release. DM makes no such guarantee when it comes to adding new fields.

Reply:
The point is that if something changes, we need to understand what that change is and ensure it works. We shouldn't simply let things float.